prow/spyglass/viewers/metadata/metadata_viewer.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package metadata provides a metadata viewer for Spyglass
+package metadata
+
+import (
+	"bytes"
+	"encoding/json"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+const (
+	name     = "metadata-viewer"
+	title    = "Metadata"
+	priority = 0
+)
+
+func init() {
+	viewers.RegisterViewer(name, viewers.ViewMetadata{
+		Title:    title,
+		Priority: priority,
+	}, ViewHandler)
+}
+
+// Started is used to mirror the started.json artifact
+type Started struct {
+	TimestampRaw int64             `json:"timestamp"`
+	Timestamp    time.Time         `json:"-"`
+	Node         string            `json:"node"`
+	Repos        map[string]string `json:"repos"`
+	Pull         string            `json:"pull"`
+}
+
+// Finished is used to mirror the finished.json artifact
+type Finished struct {
```

Comment:
You're going to want `omitempty` on everything but the most basic -- otherwise you're going to create #6092 for your viewer. Your downstream logic for handling these should handle empty values as well.